backend/LcmCrdt.Tests/LexboxApiTests.cs

```diff
@@ -336,9 +337,66 @@ public async Task UpdateSense()
         updatedSense.Definition.Values["en"].Should().Be("updated");
     }
 
+    [Fact]
+    public async Task CreateSense_WontCreateMissingDomains()
+    {
+        var senseId = Guid.NewGuid();
+        var createdSense = await _api.CreateSense(_entry1Id, new Sense()
+        {
+            Id = senseId,
+            SemanticDomains = [new SemanticDomain() { Id = Guid.NewGuid(), Code = "test", Name = new MultiString() }],
+        });
+        createdSense.Id.Should().Be(senseId);
+        createdSense.SemanticDomains.Should().BeEmpty("because the domain does not exist (or was deleted)");
+    }
+
+
+    [Fact]
+    public async Task CreateSense_WillCreateWithExistingDomains()
+    {
+        var senseId = Guid.NewGuid();
+        var semanticDomainId = Guid.NewGuid();
+        await DataModel.AddChange(Guid.NewGuid(), new CreateSemanticDomainChange(semanticDomainId, new MultiString() { { "en", "test" } }, "test"));
+        var semanticDomain = await DataModel.GetLatest<Objects.SemanticDomain>(semanticDomainId);
+        ArgumentNullException.ThrowIfNull(semanticDomain);
+        var createdSense = await _api.CreateSense(_entry1Id, new Sense()
+        {
+            Id = senseId,
+            SemanticDomains = [semanticDomain],
+        });
+        createdSense.Id.Should().Be(senseId);
+        createdSense.SemanticDomains.Should().ContainSingle(s => s.Id == semanticDomainId);
+    }
+
+    [Fact]
+    public async Task CreateSense_WontCreateMissingPartOfSpeech()
+    {
+        var senseId = Guid.NewGuid();
+        var createdSense = await _api.CreateSense(_entry1Id,
+            new Sense() { Id = senseId, PartOfSpeech = "test", PartOfSpeechId = Guid.NewGuid(), });
+        createdSense.Id.Should().Be(senseId);
+        createdSense.PartOfSpeechId.Should().BeNull("because the part of speech does not exist (or was deleted)");
+    }
+
+    [Fact]
+    public async Task CreateSense_WillCreateWthExistingPartOfSpeech()
+    {
+        var senseId = Guid.NewGuid();
+        var partOfSpeechId = Guid.NewGuid();
+        await DataModel.AddChange(Guid.NewGuid(), new CreatePartOfSpeechChange(partOfSpeechId, new MultiString() { { "en", "test" } }));
+        var partOfSpeech = await DataModel.GetLatest<Objects.PartOfSpeech>(partOfSpeechId);
+        ArgumentNullException.ThrowIfNull(partOfSpeech);
+        var createdSense = await _api.CreateSense(_entry1Id,
+            new Sense() { Id = senseId, PartOfSpeech = "test", PartOfSpeechId = partOfSpeechId, });
+        createdSense.Id.Should().Be(senseId);
+        createdSense.PartOfSpeechId.Should().Be(partOfSpeechId, "because the part of speech does  exist");
+    }
+
     [Fact]
     public async Task UpdateSensePartOfSpeech()
     {
+        var partOfSpeechId = Guid.NewGuid();
+        await DataModel.AddChange(Guid.NewGuid(), new CreatePartOfSpeechChange(partOfSpeechId, new MultiString() { { "en", "Adverb" } }));
```

Comment:
I'm not sure this test *should* pass as written: there are tests in liblcm to verify that parts of speech have the correct GUIDs, as defined by the master list in GOLDEtic.xml. For example, an adverb *must* have GUID 46e4fe08-ffa0-4c8b-bf98-2c56f38904d9 and no other GUID, or else the AllPOSTemplatePossibilityItemsMatchGoldEticStandard test will fail. I'm not sure the CreatePartOfSpeechChange API should allow GUIDs that don't appear in the master list.

I see that CreatePartOfSpeechChange does have a Predefined boolean property, so we've thought about this. But as far as I can tell from the FieldWorks source (e.g., the PartOfSpeechUi dialog), there's no option available in FieldWorks to add new parts of speech; instead, you *must* select them from one of the predefined ones.

Well, perhaps this test should pass *for now*... but then should start failing once the predefined list is complete.